app/main/register-ledger-listeners.ts

```diff
@@ -0,0 +1,151 @@
+import { ipcMain } from 'electron';
+import { BehaviorSubject, Subject, timer, combineLatest, from, of } from 'rxjs';
+import { delay, filter, map, switchMap, take } from 'rxjs/operators';
+import type Transport from '@ledgerhq/hw-transport';
+import TransportNodeHid from '@ledgerhq/hw-transport-node-hid';
+import { safeAwait } from '@blockstack/ui';
+import StacksApp, { LedgerError, ResponseVersion } from '@zondax/ledger-blockstack';
+import {
+  ledgerRequestSignTx,
+  ledgerRequestStxAddress,
+  ledgerShowStxAddress,
+} from './ledger-actions';
+
+type LedgerEvents =
+  | 'create-listener'
+  | 'remove-listener'
+  | 'waiting-transport'
+  | 'disconnected'
+  | 'has-transport';
+
+type LedgerStateEvents =
+  | Record<'name', LedgerEvents>
+  | (Record<'name', keyof typeof LedgerError> & ResponseVersion & { action: 'get-version' });
+
+interface LedgerMessageType {
+  type: 'ledger-event';
+}
+
+export type LedgerMessageEvents = LedgerStateEvents & LedgerMessageType;
+
+const POLL_LEDGER_INTERVAL = 1_250;
+const SAFE_ASSUME_REAL_DEVICE_DISCONNECT_TIME = 2_000;
+
+const ledgerState$ = new Subject<LedgerStateEvents>();
+const ledgerDeviceBusy$ = new BehaviorSubject(false);
+const listeningForDevice$ = new BehaviorSubject(false);
+const checkDisconnect$ = new Subject<void>();
+
+const transport$ = new BehaviorSubject<Transport | null>(null);
```

Comment:
Curious about the `name$` convention, is it to indicate a rxjs type?

Reply:
Yeah it's a pattern often followed to indicate something is an Observable. Finnish notation https://github.com/cartant/eslint-plugin-rxjs/blob/main/docs/rules/finnish.md